packages/web/src/pages/app/videos.tsx

```diff
@@ -13,23 +13,31 @@ export interface Video {
 
 const Videos: NextPage = () => {
   const [videos, setVideos] = React.useState<Video[]>([]);
+  const [searchQuery, setSearchQuery] = React.useState('');
+  const currentSearchQuery = React.useRef(searchQuery);
+  currentSearchQuery.current = searchQuery;
 
   React.useEffect(() => {
     const fetchVideos = async () => {
-      // Get all videos
-      const res = await fetch('/api/videos');
+      // Get videos 
+      const res = await fetch(`/api/videos?q=${searchQuery}`);
       const videosList = await res.json();
 
       // Set page status
-      setVideos(videosList);
+      if(currentSearchQuery.current===searchQuery){
+        setVideos(videosList);        
+      }
     };
 
     fetchVideos();
-  }, []);
-
+  }, [searchQuery]);
+  
   return (
     <AppLayout>
-      <Heading>Videos</Heading>
+      <HStack>
+        <Heading flex={1} >Videos</Heading>
```

Comment:
What's the purpose of the `flex={1}`? I imagine there's a better way to do this...